blaze/tests/test_interactive.py

```diff
@@ -94,6 +95,13 @@ def test_repr():
     assert '...' in result
 
 
+def test_str_does_not_repr():
+    # see GH issue #1240.
+    d = Data([('aa', 1), ('b', 2)], dshape='2 * {a: string, b: int64}')
+    expr = transform(d, c=d.a.strlen() + d.b)
+    assert sub(r'_\d+', 'XXX', str(expr)) == "Merge(_child=XXX, children=(XXX, label(strlen(_child=XXX.a) + XXX.b, 'c')))"
```

Comment:
small nitpick can you keep the `re.sub` call style?

i also think that even though the name of the expression is an implementation detail it should be consistently numbered and therefore it'd be okay to not do the replace and just keep `_1` or whatever it happens to be